InteractiveHtmlBom/web/render.js

```diff
@@ -342,16 +342,34 @@ function drawFootprints(canvas, layer, scalefactor, highlight) {
   ctx.lineWidth = 3 / scalefactor;
   var style = getComputedStyle(topmostdiv);
   var padcolor = style.getPropertyValue('--pad-color');
+  var highlightedPadColor = style.getPropertyValue('--pad-color-highlight');
+  var darkenedPadColor = style.getPropertyValue('--pad-color-highlight-darkened');
+  var bothPadColor = style.getPropertyValue('--pad-color-highlight-both');
   var outlinecolor = style.getPropertyValue('--pin1-outline-color');
-  if (highlight) {
-    padcolor = style.getPropertyValue('--pad-color-highlight');
-    outlinecolor = style.getPropertyValue('--pin1-outline-color-highlight');
-  }
+  var highlightedOutlinecolor = style.getPropertyValue('--pin1-outline-color-highlight');
+  var darkenedOutlinecolor = style.getPropertyValue('--pin1-outline-color-highlight-darkened');
+  var bothOutlinecolor = style.getPropertyValue('--pin1-outline-highlight-both');
+
   for (var i = 0; i < pcbdata.footprints.length; i++) {
     var mod = pcbdata.footprints[i];
     var outline = settings.renderDnpOutline && pcbdata.bom.skipped.includes(i);
-    if (!highlight || highlightedFootprints.includes(i)) {
+    if (!highlight) {
       drawFootprint(ctx, layer, scalefactor, mod, padcolor, outlinecolor, highlight, outline);
+    } else {
+      if(highlightedFootprints.includes(i)) {
+        if(darkenedFootprints.has(i)) {
+          drawFootprint(ctx, layer, scalefactor, mod, bothPadColor, bothOutlinecolor, highlight, outline);
+        } else {
+          drawFootprint(ctx, layer, scalefactor, mod, highlightedPadColor, highlightedOutlinecolor, highlight, outline);
+        }
+      } else {
+        if(darkenedFootprints.has(i)) {
+          console.log("darkened " + darkenedPadColor + "  plain " + padcolor);
+          drawFootprint(ctx, layer, scalefactor, mod, darkenedPadColor, darkenedOutlinecolor, highlight, outline);
+        } else {
+          drawFootprint(ctx, layer, scalefactor, mod, padcolor, outlinecolor, highlight, outline);
```

Comment:
This now draws a bounding box for every single footprint which is a bug.

In general the whole block has too many nested conditions.

I suggest following: 
* make padcolor and outlinecolor a single object passed into drawFootprint()
* precalculate 4 instances of that object based on style: normal, highlighted, darkened, both.
* write non-nested if () {} else if () {} ... statements that handle each case:
  * non-highlight mode: draw normal
  * highlightedFootprints and darkenedFootprints have the ref: draw with "both" color
  * darkenedFootprints has the ref: draw "darkened"
  * highlightedFootprints has the ref: draw "highlighted"